internal/codegen/codegen.go

```diff
@@ -0,0 +1,74 @@
+package codegen
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+const hookdeckOpenAPISchemaURI = "https://api.hookdeck.com/2024-03-01/openapi"
+
+func RunCodeGen() error {
+	fmt.Println("generating Hookdeck source verifications")
+
+	// Load OpenAPI schema
+	loader := openapi3.NewLoader()
+	u, err := url.Parse(hookdeckOpenAPISchemaURI)
+	if err != nil {
+		return err
+	}
+	fmt.Println("getting Hookdeck OpenAPI schema")
+	doc, err := loader.LoadFromURI(u)
+	if err != nil {
+		return err
+	}
+
+	// Get list of source verification providers
+	verificationListAny, err := doc.Components.Schemas.JSONLookup("VerificationConfig")
+	if err != nil {
+		return err
+	}
+	verificationList, ok := verificationListAny.(*openapi3.Schema)
+	if !ok {
+		return errors.New("type assertion failed")
+	}
+
+	// Construct verification data
+	verifications := []Verification{}
+	for _, schemaRef := range verificationList.OneOf {
+		// SDK not supported yet
+		if schemaRef.Ref == "#/components/schemas/VerificationMailchimp" || schemaRef.Ref == "#/components/schemas/VerificationPaddle" {
+			continue
+		}
```

Comment:
The Hookdeck Go SDK v0.4.1 hasn't supported Mailchimp & Paddle yet, so I have to add this to avoid generating those providers.

Reply:
I can see this being a bigger issue in the future where the OpenAPI schema changes and the Go SDK hasn't caught up. We may need to refactor the codegen logic to be more declarative about which providers we support.

Reply:
Maybe we trigger a code gen on new GO SDK version, that should fix it?

Reply:
No, the breaking change is the other way around where there's a new provider but it's not supported in the Go SDK yet. Because this codegen is from the schema and not from the SDK, we assume that it's already supported, so if it's not it will cause issue here.

Reply:
Yes, sure, but I'm saying we set up a GH action to run the codegen on the Go SDK version. If the Go SDK versions has been bumped then the OpenAPI schema also has to have the new provider. Technically you could still have a race condition but that seems unlikely enough.

Reply:
I think it depends on when the SDK is updated.

Let's say we release the server that supports a new provider today. When is this provider supported in the SDK? Let's say it takes 1 day (because we forget to make a release).

During this time, we can't run codegen on the Terraform provider, which also blocks documentation generation.

I don't think this is unlikely, given Mailchimp & Paddle are not supported yet. So either we have to establish new processes around Go SDK generation, or we should improve this codegen to prevent this blocker case.

Reply:
The provider should be supported in the SDK more or less right away since the release is automated. cc @leggetter please confirm

Reply:
We don't fully automate anything yet. There's always a manual release step, so we can:

- Check the PR
- Manually check and update the Release notes
- Perform the release and determine the semantic version
- Distribute the release

This may make it sound more complicated than it is.

1. Open API spec creates PR
2. Review and merge API spec
3. Generate PR for TypeScript SDK, review, merge, release, socialize
4. Generate PR for Go SDK, review, merge, release, socialize
5. Generate PR for Terraform Provider, review, merge, release, socialize

I think this is the correct approach right now because we need manual intervention for the release due to the SemVer, to create useful release notes, and for it to trigger the social side of things.

Reply:
Sounds good for now

Reply:
@alexluong SDK v0.5.0 is out with Mailchimp and Paddle support.